NotesApp/Scenes/List/ListViewModel.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  ListViewModel.swift
+//  NotesApp
+//
+//  Created by Luka Gujejiani on 31.10.24.
+//
+
+import Foundation
+
+final class ListViewModel {
+    // MARK: - Properties
+    private var notes: [NoteModel] = [] {
+        didSet {
+            onNotesUpdated?()
+        }
+    }
+    
+    var onNotesUpdated: (() -> Void)?
+    
+    // MARK: - Lifecycle
+    init() {
+        print("✅⬅️ Init called in ListViewModel (VIEWMODEL)")
+        fetchNotes()
+    }
+    
+    deinit {
+        print("🗑️⬅️ Deiniting ListViewModel (VIEWMODEL)")
+    }
+    
+    // MARK: - Methods
+    func fetchNotes() {
+        notes = DataManager.shared.fetchNotes()
+    }
+    
+    func addNote(title: String, body: String) {
+        DataManager.shared.addNote(title: title, body: body)
+        fetchNotes()
+    }
+    
+    func deleteNote(at index: Int) {
+        let note = notes[index]
+        DataManager.shared.deleteNote(note)
+        fetchNotes()
+    }
+    
+    func noteCount() -> Int {
+        return notes.count
+    }
```

Comment:
I would prefer
 var noteCount: Int {
    notes.count
} 
func îs better if you have calculation or more complex thing for example you are passing something and calculating number but in this case having variable has same result but it is up to you you can research which one is better in this case and why